drivers/ieee802154/ieee802154_kw41z.c

```diff
@@ -322,6 +335,27 @@ static inline void kw41z_rx(struct kw41z_context *kw41z, u8_t len)
 			frag->data[i] = reg_val >> 8;
 		}
 	}
+#else /* CONFIG_SOC_MKW40Z4 */
+	u32_t reg_val = 0;
+
+	/* PKT_BUFFER_RX needs to be accessed alligned to 32 bits */
+	for (i = 0; i < pkt_len; i++) {
+		switch (i % 4) {
+		case 0:
+			reg_val = *(((u32_t *)ZLL->PKT_BUFFER) + i/4);
```

Comment:
I know you're following the pattern from KW41 above, but is there a reason you didn't implement it like this:
```
reg_val = ZLL->PKT_BUFFER[i/4];
```